Jenkinsfile

```diff
@@ -29,16 +35,21 @@ try {
         }
 
         // Toggles for PR vs. Master builds.
-        // For PR's, we just build for arm-v7a and run unit tests for the ObjectServer variant
-        // A full build is done on `master`.
-        // TODO Once Android emulators are available on all nodes, we can switch to x86 builds
-        // on PR's for even more throughput.
+        // - For PR's, we favor speed > absolute correctness. So we just build for x86, use an
+        //   emulator and run unit tests for the ObjectServer variant.
+        // - For branches from which we make releases, we build all architectures and run tests
+        //   on an actual device.
+        def useEmulator = false
+        def emulatorImage = ""
         def abiFilter = ""
         def instrumentationTestTarget = "connectedAndroidTest"
+        def deviceSerial = ""
         if (!releaseBranches.contains(currentBranch)) {
-          abiFilter = "-PbuildTargetABIs=armeabi-v7a"
+          useEmulator = true
+          emulatorImage = "system-images;android-29;default;x86"
+          abiFilter = "-PbuildTargetABIs=x86"
           instrumentationTestTarget = "connectedObjectServerDebugAndroidTest"
-          // Run in debug more for better error reporting
+          deviceSerial = "emulator-5554"
```

Comment:
Where does this come from? Assumes it is built into the emulator image.

Reply:
Yes, it is the default serial id.